File: map/lab8/lab8/repository/IRepository.cs
using System.Collections.Generic;
using lab8.domain;

namespace lab8.repository
{
    public interface IRepository<in TId, TEntity>
        where TEntity : Entity<TId>
        where TId : class
    {
        TEntity Get(TId id);
        List<TEntity> Get();
        void Add(TEntity entity);
        void Update(TEntity entity);
        void Delete(TId id);
    }
}

File: mpp/lab3/src/main/java/repository/IRepository.java
package repository;

import java.util.List;

public interface IRepository<E> {
    void add(E e) throws RepositoryError;
    List<E> find() throws RepositoryError;
    E findById(E e) throws RepositoryError;
    void update(E e) throws RepositoryError;
    void delete(E e) throws RepositoryError;
}
